lib/autoload/duplicator.rb

```diff
@@ -41,6 +41,18 @@ def duplicate(stuff)
     stuff_is_enumerable ? duplicated_stuff : duplicated_stuff[0]
   end
 
+  # Time shifts the datetime passed to this function.
+  # Cap the maximum datetime to 31 December 9999 0:00:00 as the frontend JS cannot support
+  # 5 digit years.
+  #
+  # @param [DateTime] original_time The DateTime value to be time shifted.
+  # @return [DateTime] The time shifted DateTime, capped to 31 December 9999.
+  def time_shift(original_time)
+    max_time = Time.zone.local(9999, 12, 31, 0, 0, 0)
```

Comment:
Let's leave it here first but add a TODO on this to figure out how better to do this?